### ISSUE TITLE
Add support for directly using path to binaries

### ISSUE THREAD

**Issue by guriandoro** [datacharmer/dbdeployer#21]
Hi Guiseppe!

First of all, thanks for creating another great tool :)

**Is your feature request related to a problem? Please describe.**
We use /opt/ (`/opt/mysql/`, `/opt/percona-server/`, etc) to store binaries in a shared-user testing server (mainly to avoid wasting disk space by storing repeated binaries in home directories). In this case, the only way to run `dbdeployer` seems to be by using the `--sandbox-binary` argument, like:

```
dbdeployer --sandbox-binary=/opt/mysql/ deploy single 8.0.4
```

This works ok, but what I'm lacking (that we had with `mysql_sandbox`) is the ability to "find" binaries while writing the command (by using Tab autocompletion on the path), like:

```
dbdeployer deploy single /opt/mysql/^tab^tab
... get list of dirs under /opt/mysql/ ...
dbdeployer deploy single /opt/mysql/8.0.4
```

**Describe the solution you'd like**
To be able to use commands like:
```
dbdeployer deploy single /opt/mysql/8.0.4
```

**Describe alternatives you've considered**
Maybe using a conf file, or templating? But I haven't checked that further, until discussing with you first. Those seem like hacks that will probably not work correctly for what I explained above.

Thanks!

**Comment by datacharmer** [comment]
To define the sandbox binary folder, there are currently three methods:
* (temporarily) using the `--sandbox-binary` option;
* (temporarily) using the environment variable `SANDBOX_BINARY`;
* (permanently) changing the field `sandbox-binary` in the defaults, using `dbdeployer defaults update sandbox-binary /path/to/binaries`.

When I designed dbdeployer, I left out intentionally the ability of pointing to a directory containing the binaries. It would be easy to enhance it, and it would work well if we had
`dbdeployer deploy single $HOME/opt/percona/5.7.21`

There would be, however, a few drawbacks:
1. a command like `dbdeployer deploy single $HOME/opt/percona/latest_build` needs to be supplemented with the server version (`--binary-version`) but then the _easy_ enhancement mentioned above would not be easy after all.
2. if you install the same version from two different paths, you will have a conflict in the sandbox name, and thus you would again need to use an option (`--sandbox-directory`).

The ability of using tab for name completion will probably cost you extra options in some very common cases as mentioned above.

The recommended organization for binaries is to use prefixes.
Instead of

    $HOME/opt/mysql/5.7.21
    $HOME/opt/percona/5.7.21

you could use 

    $HOME/opt/mysql/5.7.21
    $HOME/opt/mysql/ps5.7.21

The advantage of this method (which works out of the box) is that you can now install both `5.7.21` and `ps5.7.21` and dbdeployer would install both without need for additional options: the sandbox directory would be different because of the version, and the port is automatically calculated using the first available one.

That said, if none of the suggested methods works for you, I can implement the change, although it may take a while to make it strong enough to make it fool proof.

**Comment by guriandoro** [comment]
Ok, thanks! Let me discuss it with the team, to see what will work better for all of us, and I'll write back.

I will most likely reply back next week, or the other, at most.

PS: I can help with coding or testing, if needed, if it comes to that.

**Comment by datacharmer** [comment]
Implemented in version 1.8.0

**Comment by guriandoro** [comment]
Awesome! Great news :) Works like a charm:

```
agustin@bm-support01 ~/bin $ ./dbdeployer deploy single /opt/mysql/8.0.4/
Database installed in $HOME/sandboxes/msb_8_0_4
run 'dbdeployer usage single' for basic instructions'
.. sandbox server started
agustin@bm-support01 ~/bin $ $HOME/sandboxes/msb_8_0_4/use
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 8
Server version: 8.0.4-rc-log MySQL Community Server (GPL)
...
```

Thanks again, Giuseppe.
